cmd/snap-update-ns/change.go

```diff
@@ -498,7 +498,13 @@ func neededChangesImpl(currentProfile, desiredProfile *osutil.MountProfile) []*C
 		if reuse[current[i].Dir] {
 			changes = append(changes, &Change{Action: Keep, Entry: current[i]})
 		} else {
-			changes = append(changes, &Change{Action: Unmount, Entry: current[i]})
+			var entry osutil.MountEntry = current[i]
+			entry.Options = append([]string(nil), entry.Options...)
+			// Detach directory bind mounts and filesystem mounts.
+			if (entry.Type == "tmpfs" || entry.OptBool("bind") || entry.OptBool("rbind")) && !entry.XSnapdDetach() {
```

Comment:
Nested condition makes it a bit hard to read, maybe organizing the code like this would make it clearer:
```
needsDetach := entry.Type == "tmpfs" || entry.OptBoot("bind") || entry.OptBool("rbind")
if needsDetach && !enrty.XSnapdDetach() {
   ...
}
```